### PR TITLE
Directory component Mentee

### PR DESCRIPTION
**Pull request by 21shield**
### Ticket
closes #  Create Directory Component
[Trello Card](https://trello.com/c/1zdguhSq/51-create-directory-component)
### Checklist
- [x] There are no warnings in the browser console
- [x] I made sure the component is WCAG compliant using ANDI
- [x] I made sure the UI follows the style guide in the wireframe (font, font-size, etc...)
- [x] I made sure that the screen reader can read all the text
- [x] I made sure I removed all the console.logs
- [x] I made sure mobile portrait view looks good
- [x] I made sure mobile landscape view looks good
- [x] I added storybook for the component
	- [ ] Storybook not applicable
- [x] I added unit test coverage
	- [ ] I added integration test coverage
	- [ ] Unit test coverage not applicable

### Browser Views 
 Mobile | IPad | fullScreen | Search Not Found | 
------------ | -------------| -------------| -------------
Screen Shot 2021-04-20 at 3 36 16 PM  |  Screen Shot 2021-04-20 at 3 36 34 PM |  Screen Shot 2021-04-20 at 3 37 09 PM |  Screen Shot 2021-04-20 at 3 37 23 PM |
### Tests
 EmptyDirectory  | MentorCard  | DirectorySearchBar  | Directory
------------ | -------------| -------------| -------------
Screen Shot 2021-04-20 at 3 56 08 PM | Screen Shot 2021-04-20 at 3 56 34 PM |  Screen Shot 2021-04-20 at 3 54 49 PM |  Screen Shot 2021-04-20 at 4 04 40 PM

### CODE REVIEW

**Comment by 21shield** [comment]
## Wrong Font styles
Updated Font Styles in ## 02ffaab

File: src/components/Directory/index.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  const [mentorCards, setMentorCards] = useState([]);
+  const [query, setQuery] = useState('');
+
+  const handleChange = (e, newVal) => setSelectedTab(newVal);
+
+  useEffect(() => {
+    async function fetchData() {
+      const { data } = await request.get('/api/directory');
+      setMentorCards(data);
+    }
```
Comment (cfernando415) [inline src/components/Directory/index.js:33]: we need to handle error here; let's do a try/catch block, and set a state for error, if error then throw new Error()

File: src/components/Directory/utils.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
```
Comment (cfernando415) [inline src/components/Directory/utils.js:1]: i don't think we need this here because we're not using jsx here

**Review by cfernando415 — CHANGES_REQUESTED** [review]
pls see comments

File: src/components/Directory/utils.js (hunk 2 above)
Comment (21shield) [reply src/components/Directory/utils.js:1]: When I comment out this import I get this error
Screen Shot 2021-04-23 at 1 53 16 PM

File: src/components/Directory/DirectorySearch/index.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import { TextField, InputAdornment, useMediaQuery } from '@material-ui/core';
+import Button from '@material-ui/core/Button';
```
Comment (PeaWarrior) [inline src/components/Directory/DirectorySearch/index.js:4]: Can import from same module

File: src/components/Directory/MentorCard/index.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+import PropTypes from 'prop-types';
+import useStyles from './ MentorCard.styles';
+import Card from '@material-ui/core/Card';
+import CardHeader from '@material-ui/core/CardHeader';
+import Divider from '@material-ui/core/Divider';
+import Avatar from '@material-ui/core/Avatar';
+import Button from '@material-ui/core/Button';
+import CardContent from '@material-ui/core/CardContent';
+import Typography from '@material-ui/core/Typography';
+import useMediaQuery from '@material-ui/core/useMediaQuery';
```
Comment (PeaWarrior) [inline src/components/Directory/MentorCard/index.js:11]: Combine.

File: src/components/Directory/index.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+      <DirectorySearchBar setQuery={setQuery} />
+      <div className={mentorDirectory}>
+        <Tabs
+          value={selectedTab}
+          onChange={handleChange}
+          aria-label="Mentor Directory Tabs"
+          className={tabs}
+          TabIndicatorProps={{
+            style: { background: '#550CCC' },
+          }}
```
Comment (PeaWarrior) [inline src/components/Directory/index.js:83]: Can we use ES6 for this to keep it consistent.
{...{
  value: selectedTab,
  onChange: handleChange,
  ...
}}

File: src/components/Directory/MentorCard/index.js (hunk 4 above)
Comment (21shield) [reply src/components/Directory/MentorCard/index.js]: Resolved in : 193e9b4

File: src/components/Directory/index.js (hunk 5 above)
Comment (21shield) [reply src/components/Directory/index.js]: Resolved in : 754213b

:D

File: src/components/Directory/utils.js (hunk 2 above)
Comment (21shield) [reply src/components/Directory/utils.js]: Resolved by changing location of SubheaderIcon : 17ee558

File: src/components/Directory/index.js (hunk 1 above)
Comment (21shield) [reply src/components/Directory/index.js]: Resolved in : 947eb29

Comment (21shield) [reply src/components/Directory/index.js]: Added a loading component when fetching data in commit # 3eb35d5

Comment (21shield) [reply src/components/Directory/index.js]: Updated conditional statement for directory on load commit: 13938f9

File: src/components/Directory/MentorCard/index.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    skillList,
+    jobField,
+    mentorNameField,
+    bioSection,
+  } = classes;
+
+  const handleClick = e => {
+    e.preventDefault();
+    if (onTestClick) return onTestClick();
+    console.log('BOOKING FOR THIS MENTOR OPENS UP');
```
Comment (cfernando415) [inline src/components/Directory/MentorCard/index.js:39]: let's remove the console.log and replace it with // TODO: Booking for this mentor opens up

**Review by cfernando415 — CHANGES_REQUESTED** [review]
pls see comment

File: src/components/Directory/index.js (hunk 1 above)
Comment (cfernando415) [reply src/components/Directory/index.js]: resolving this convo :+1:

File: src/components/Directory/MentorCard/index.js (hunk 6 above)
Comment (21shield) [reply src/components/Directory/MentorCard/index.js]: Resolved in commit:  a6ddf95
* Changed console to a comment that reads for next stage

**Review by cfernando415 — APPROVED** [review]
:+1:

**Pull request merged**